packages/graphql/src/translate/queryAST/ast/fields/attribute-fields/CypherUnionAttributePartial.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ConcreteEntityAdapter } from "../../../../../schema-model/entity/model-adapters/ConcreteEntityAdapter";
+import type { QueryASTContext } from "../../QueryASTContext";
+import { QueryASTNode } from "../../QueryASTNode";
+import type { Field } from "../Field";
+import Cypher from "@neo4j/cypher-builder";
+
+export class CypherUnionAttributePartial extends QueryASTNode {
```

Comment:
Well, this is one of the workaround done to support unions as we have at the moment.

Essentially, a UnionAttribute is a wrapper over multiple Attributes, each one of those we decided to call Partial (effectively, one of the concrete instances of an union)

I don't think the name is wrong, but this needs to be covered in some document

Reply:
even a code comment could help

Reply:
It's worth noting that the name "partial" is also present in the Composite{Read/Connection} classes. I personally think this could be solved by some internal documentation!